tooling/samples/index.js

```diff
@@ -39,8 +39,18 @@ const primitive = schema => {
 
 const sampleFromSchema = (schema, config = {}) => {
   const objectifySchema = objectify(schema);
-  let { type } = objectifySchema;
-  const { example, properties, additionalProperties, items } = objectifySchema;
+  let { type, properties, items } = objectifySchema;
+  const { example, additionalProperties } = objectifySchema;
+
+  const usesPolymorphism = hasPolymorphism(objectifySchema);
+  if (usesPolymorphism === 'allOf') {
+    const mergedAllOf = mergeAllOf(objectifySchema);
+    properties = mergedAllOf.properties;
+    items = mergedAllOf.items;
+  } else if (usesPolymorphism) {
+    properties = objectifySchema[usesPolymorphism][0].properties;
+    items = objectifySchema[usesPolymorphism][0].items;
+  }
```

Comment:
Weird, we already have handling for `allOf` a few lines down https://github.com/readmeio/oas/blob/master/tooling/samples/index.js#L61-L74

How come that code wasn't getting picked up?

Reply:
Ah, if I turn:
```
"reach": {
  "allOf": [
    {
      "$ref": "#/components/schemas/SegmentReach"
    }
  ],
  "type": "object"
},
```
to
```
"reach": {
  "allOf": [
    {
      "$ref": "#/components/schemas/SegmentReach"
    }
  ]
},
```
it correctly renders.

Reply:
Interesting! I looked around and the presence of a type property as a sibling to a ref (or an all of) is completely valid.

So we need to support this!

Maybe we can consolidate this logic with the existing allOf logic though?

Reply:
Yeah after looking at this a little closer I think it might be more clear what's going on if when we detect polymorphism we recurse into that object instead of overwriting `items` and `properties`, similar to how we're already handling `allOf` when there's no `type` present. Something like:

```js
if (hasPolymorphism) {
  if (hasPolymorphism === 'allOf') {
    try {
      return sampleFromSchema(
        mergeAllOf(objectifySchema, {
          resolvers: {
            // Ignore any unrecognized OAS-specific keywords that might be present on the schema (like `xml`).
            defaultResolver: mergeAllOf.options.resolvers.title,
          },
        })
      );
    } catch (err) {
      // Unable to merge the schema for whatever reason so let's just ignore it and do a no-op here.
      return undefined
    }
  } else {
    return sampleFromSchema(objectifySchema[hasPolymorphism]);
  }
}
```

Reply:
updated!

Reply:
I don't think we need this and the allOf handling from line 81 to 94. Worth testing, and deleting if we can.

Reply:
When I initially wrote this, I thought this portion would handle the `anyOf` and `oneOf` schemas with `type` cases, but I didn't correctly test it. I updated it and confirmed it works for those cases and added a test for it.